.travis.yml

```diff
@@ -2,38 +2,57 @@
 language: node_js
 node_js:
   - "6"
-  - "8"
-  - "10"
 
 addons:
   chrome: stable
 
 cache:
-  directories:
-    - $HOME/.npm
-
-env:
-  - TEST_FRAMEWORK=ember-qunit
-  - TEST_FRAMEWORK=ember-cli-mocha
-
-matrix:
-  fast_finish: true
-
-branches:
-  only:
-    - master
+  yarn: true
+
+jobs:
+  fail_fast: true
+  allow_failures:
+    - env: EMBER_TRY_SCENARIO=ember-canary TEST_FRAMEWORK=ember-qunit
+    - env: EMBER_TRY_SCENARIO=ember-canary TEST_FRAMEWORK=ember-cli-mocha
+
+  include:
+    # runs linting and tests with current locked deps
+
+    - stage: "Tests"
+      name: "Tests"
+      install:
+        - yarn install --non-interactive
+        - ./bin/install-test-framework.sh
+      script:
+        - yarn lint:js
+        - yarn test
+
+    - name: "Floating Dependencies"
+      script:
+        - yarn test
+
+    # we recommend new addons test the current and previous LTS
+    # as well as latest stable release (bonus points to beta/canary)
+    - stage: "Additional Tests"
+    - env: EMBER_TRY_SCENARIO=ember-lts-2.16 TEST_FRAMEWORK=ember-qunit
```

Comment:
```suggestion
      env: EMBER_TRY_SCENARIO=ember-lts-2.16 TEST_FRAMEWORK=ember-qunit
```

Reply:
magik?

Reply:
I see this is pretty surprising `.travis.yml` behavior....

```
- stage: "Additional Tests"
- env: FOO=bar
```

is actually 2 jobs:

1. `- stage: "Additional Tests"`
2. `- env: FOO=bar`

Reply:
the following `- env...` will be other jobs, but belong to same stage as the first 
```
- stage: ''
  env:
```